plutus-pab/src/Cardano/Wallet/Server.hs

```diff
@@ -98,25 +98,24 @@ app nodeClientEnv chainIndexEnv mVarState =
     (submitTxn >=> const (pure NoContent)) :<|> ownPubKey :<|> uncurry updatePaymentWithChange :<|>
     walletSlot :<|> ownOutputs
 
-main :: MonadIO m => Config -> BaseUrl -> BaseUrl -> Availability -> m ()
-main Config {baseUrl, wallet} nodeBaseUrl chainIndexBaseUrl availability = runStdoutLoggingT $ do
-    let port = baseUrlPort baseUrl
-        state = emptyWalletState wallet
-    nodeClientEnv <-
-        liftIO $ do
-            nodeManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv nodeManager nodeBaseUrl
-    chainIndexEnv <-
-        liftIO $ do
-            chainIndexManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv chainIndexManager chainIndexBaseUrl
+main :: Trace IO WalletMsg -> Config -> NodeUrl -> ChainIndexUrl -> Availability -> IO ()
```

Comment:
Hmm, not sure what our policy should be on `MonadIO` vs `IO`, e.g. there are some other `MonadIO` things in this module. I'd be fine with mostly fixing IO, I think?

Reply:
yeah sounds good to me. I'll do that

Reply:
Actually - on second thought: i want to keep these PRs "minimally invasive' while i get to know the code base better. I do note all your comments though and *will* get back to addressing them! Thanks!